soc/st/stm32/Kconfig.defconfig

```diff
@@ -85,6 +85,14 @@ config FLASH_BASE_ADDRESS
 		if $(DT_FLASH_PARENT_IS_XSPI)
 	default $(dt_chosen_reg_addr_hex,$(DT_CHOSEN_Z_FLASH))
 
+# Allows the SoC clock driver to correctly initialize the
+# Q/O/XSPI controller clocks when the application is residing
+# in external Flash and is chainloaded with MCUboot.
+# Whether the app is eXecuted in Place (XiP) depends on the MCUboot mode used.
+config STM32_APP_IN_EXT_FLASH
```

Comment:
I think the config should be defined in **Kconfig** with the description in the `help` section.
This **Kconfig.defconfig** file would only set its default value.